docs/authentication.md

```diff
@@ -303,3 +303,141 @@ if ($user->tokenCant('forums.manage')) {
     // do something....
 }
 ```
+
+## HMAC SHA256 Token Authenticator
+
+The HMAC-SHA256 authenticator supports the use of revoke-able API keys without using OAuth. This provides
+an alternative to a token that is passed in every request and instead uses a shared secret that is used to sign
+the request in a secure manner. Like authorization tokens, these are commonly used to provide third-party developers
+access to your API. These keys typically have a very long expiration time, often years.
+
+These are also suitable for use with mobile applications. In this case, the user would register/sign-in
+with their email/password. The application would create a new access token for them, with a recognizable
+name, like John's iPhone 12, and return it to the mobile application, where it is stored and used
+in all future requests.
+
+> **Note** For the purpose of this documentation and to maintain a level of consistency with the Authorization Tokens,
+the term "Token" will be used to represent a set of API Keys (key and secretKey).
+
+### Usage
+In order to use HMAC Keys/Token the `Authorization` header will be set to the following in the request:
+
+```
+Authorization: HMAC-SHA256 <key>:<HMAC HASH of request body>
+```
+
+The code to do this will look something like this:
+
+```php
+<?php
+header("Authorization: HMAC-SHA256 {$key}:" . hash_hmac('sha256', $requestBody, $secretKey));
```

Comment:
Can you add a usable example where `$key`, `$requestBody`, and `$secretKey` are defined. I'm fearing someone would copy-paste this example and complain it's not working.

Reply:
CI4 users should not use `header()` at all. Or at least I think it is not good practice.
If you fear someone would copy-paste, the code should use the Response class?

Reply:
@paulbalandan I like that idea and will add a tangible example shortly.

@kenjis I used `header()` here as it is intended to be a generic reference for a third-party app calling this API.  I felt that a CI4 CURLRequest class reference here may be confusing as to the purpose of this code sample.  If you feel strongly otherwise, I will adjust it.

Reply:
I added a second example using the CURLRequest() class and showing an actual hash calculation.